community/kernel/src/main/java/org/neo4j/kernel/impl/logging/AsyncLogger.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.logging;
+
+import org.neo4j.concurrent.AsyncEventSender;
+import org.neo4j.function.Consumer;
+import org.neo4j.logging.Logger;
+
+import static org.neo4j.kernel.impl.logging.AsyncLogEvent.logEvent;
+
+/**
+ * Merely a means of capturing logging statements, converting them to {@link AsyncLogEvent log events}, which are
+ * fed into an {@link AsyncEventSender}. These events will be processed later where call to the actual logger
+ * will happen.
+ */
+class AsyncLogger implements Logger
+{
+    private final AsyncEventSender<AsyncLogEvent> sender;
+    private final Logger logger;
+
+    AsyncLogger( AsyncEventSender<AsyncLogEvent> sender, Logger logger )
+    {
+        this.sender = sender;
+        this.logger = logger;
+    }
+
+    @Override
+    public void log( String message )
+    {
+        sender.send( logEvent( logger, message ) );
```

Comment:
Obviously we should measure, but I don't think this is a problem since we don't have very much logging. The garbage of the event instances are dwarfed by the strings and exceptions themselves.